sdk/keyvault/azure-security-keyvault-secrets/test/samples/sample2-backup-restore/sample2-backup-restore.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief This sample provides the code implementation to use the Key Vault Secrets SDK client for
+ * C++ to backup, restore, delete and purge a secret.
+ *
+ * @remark The following environment variables must be set before running the sample.
+ * - AZURE_KEYVAULT_URL:  To the Key Vault account URL.
+ * - AZURE_TENANT_ID:     Tenant ID for the Azure account.
+ * - AZURE_CLIENT_ID:     The Client ID to authenticate the request.
+ * - AZURE_CLIENT_SECRET: The client secret.
+ *
+ */
+
+#if defined(_MSC_VER)
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+
+#include <azure/core.hpp>
+#include <azure/identity.hpp>
+#include <azure/keyvault/keyvault_secrets.hpp>
+
+#include <chrono>
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+using namespace Azure::Security::KeyVault::Secrets;
+void AssertSecretsEqual(Secret const& expected, Secret const& actual);
+
+int main()
+{
+  auto tenantId = std::getenv("AZURE_TENANT_ID");
+  auto clientId = std::getenv("AZURE_CLIENT_ID");
+  auto clientSecret = std::getenv("AZURE_CLIENT_SECRET");
+  auto credential
+      = std::make_shared<Azure::Identity::ClientSecretCredential>(tenantId, clientId, clientSecret);
+
+  // create client
+  SecretClient secretClient(std::getenv("AZURE_KEYVAULT_URL"), credential);
+
+  std::string secretName("MySampleSecret2");
+  std::string secretValue("my secret value");
+
+  try
+  {
+    // create secret
+    secretClient.SetSecret(secretName, secretValue);
+
+    // get secret
+    Secret secret = secretClient.GetSecret(secretName).Value;
+
+    std::cout << "Secret is returned with name " << secret.Name << " and value " << secret.Value
+              << std::endl;
+
+    size_t backUpSize = 0;
+    {
+      std::cout << "\t-Backup Key" << std::endl;
+      std::vector<uint8_t> backupKey(secretClient.BackupSecret(secret.Name).Value.Secret);
```

Comment:
```suggestion
      std::vector<uint8_t> backupSecret(secretClient.BackupSecret(secret.Name).Value.Secret);
```

mm,, if you ask me, this is a copy-construct that is not very usual.  We are discarding the entire response+model to only keep the Secret. Can we just do:

auto backupSecret = secretClient.BackupSecrect(secret.Name).Value;
auto const& secret = backupSecret.Secret;  

Then we are telling people we can keep the backupSecret in the heap and read the secret from it directly without making a copy.